### PR TITLE
Add main peaks' shadow for event shadow

### PR DESCRIPTION
**Pull request by dachengx**
## What does the code in this PR do / what does it improve?

Specify the main peaks' shadow because later some cuts may implement on S1 or S2 peaks. 

## Can you briefly describe how it works?

Find the main S1 and S2 peaks and save them in new fields: `s1_shadow` and `s2_shadow`. This PR does not change the definition of already existing fields. 

## Can you give a minimal working example (or illustrate with a figure)?

[Note](https://xe1t-wiki.lngs.infn.it/doku.php?id=xenon:xenonnt:kexin:sr0_shadow_cut)

_Please include the following if applicable:_
  - [x] _Update the docstring(s)_
  - [x] _Update the documentation_
  - [x] _Tests to check the (new) code is working as desired._
  - [ ] _Does it solve one of the open issues on github?_

### CODE REVIEW

**Comment by kdund** [comment]
Is this required for the version if the cut in cutax or can that be added before?

**Comment by dachengx** [comment]
> Is this required for the version if the cut in cutax or can that be added before?

Yes, it is.

File: straxen/plugins/event_shadow.py [hunk 1]
```diff
@@ -9,16 +9,19 @@ class EventShadow(strax.Plugin):
...
-    __version__ = '0.0.7'
+    __version__ = '0.0.8'
     depends_on = ('event_basics', 'peak_basics', 'peak_shadow')
     provides = 'event_shadow'
     save_when = strax.SaveWhen.EXPLICIT
 
     def infer_dtype(self):
-        dtype = [('shadow', np.float32, 'previous s2 shadow [PE/ns]'),
+        dtype = [('s1_shadow', np.float32, 'main s1 shadow [PE/ns]'),
+                 ('s2_shadow', np.float32, 'mian s2 shadow [PE/ns]'),
```
Comment (kdund) [inline straxen/plugins/event_shadow.py:23]: mian->main

**Review by kdund — APPROVED** [review]
after typo-correction, I approve :)

File: straxen/plugins/event_shadow.py (hunk 1 above)
Comment (dachengx) [reply straxen/plugins/event_shadow.py]: Thanks. I just corrected it.

**Review by WenzDaniel — APPROVED** [review]
Looks fine, we should see after a while if we really need all three "shadow" values. I still have to get used to the meaning of shadow here. For me a shadow is always shaded on something, but not a property of the thing itself. In your definition it is somewhat both (large peaks casts shadow and other peaks within the shadow have the property shadow).

**Pull request merged**